js/controller.js

```diff
@@ -529,50 +549,29 @@ function controller(OO, _, $) {
       this.renderSkin({ cast: { connected: false, device: '' } });
     },
 
-    addAirPlayListeners(videoElement) {
-      if (!window.WebKitPlaybackTargetAvailabilityEvent) {
-        return;
-      }
-      const airPlayState = window.sessionStorage.getItem('airPlayState');
-      this.airPlayWasConnected = airPlayState === CONSTANTS.AIRPLAY_STATE.CONNECTED;
-      if (this.state.isAirplayAllowed) {
-        videoElement.addEventListener(
-          'webkitplaybacktargetavailabilitychanged',
-          _.bind(this.airPlayListener, this)
-        );
+    onAirplayAvailabilityChanged(eventName, isAvailable) {
+      this.state.airplay.available = isAvailable;
+      this.renderSkin();
+    },
 
-        // This event fires when a media element starts or stops AirPlay playback
-        videoElement.addEventListener(
-          'webkitcurrentplaybacktargetiswirelesschanged',
-          _.bind(this.toggleAirPlayIcon, this)
-        );
+    onAirplayConnectedChanged(eventName, isConnected) {
+      if (isConnected) {
+        this.state.airplay.statusIcon = CONSTANTS.AIRPLAY_STATE.CONNECTED;
+      } else {
+        this.state.airplay.statusIcon = CONSTANTS.AIRPLAY_STATE.DISCONNECTED;
       }
+      this.renderSkin();
     },
 
-    airPlayListener(event) {
-      this.state.isAirPlayAvailable = event.availability === 'available';
+    onAirplaySessionResumedWhenPageWasReloaded() {
```

Comment:
I think it's better to name handler close to the event name. Nothing points that it happens on page reload and looks confusing

Reply:
> I think it's better to name handler close to the event name.

okay, I thought it would be better to elaborate (for the person who would read it) since sessionResume event is fired only when page is reloaded while casting. will remove it)

> Nothing points that it happens on page reload and looks confusing

it fires only on page reload. (Named it by analogy with chromecast) Can rename it